### ISSUE TITLE
[Doc Source]: upgrade --set-args argument invalid

### ISSUE THREAD

**Issue by todeb** [openebs/mayastor#1765]
**Are you reporting an issue with existing content?**
On linked page it is an advice to use ` kubectl mayastor upgrade ` command with `--set-args` , but this argument is invalid.
Tried for both 2.7.0 and 2.7.1 kubectl-mayastor extension.

https://openebs.io/docs/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/additional-information/performance-tips

```
./kubectl-mayastor upgrade -n openebs --set-args 'io_engine.coreList={2,3}'
error: unexpected argument '--set-args' found

  tip: a similar argument exists: '--set'

Usage: kubectl-mayastor upgrade --namespace <NAMESPACE> <--registry <REGISTRY>|--allow-unstable|--dry-run|--skip-data-plane-restart|--skip-single-replica-volume-validation|--skip-replica-rebuild|--skip-cordoned-node-validation|--skip-upgrade-path-validation-for-unsupported-version|--set <SET>|--set-file <SET_FILE>>

For more information, try '--help'.
```

what is the correct command?

btw. I did:
```
kubectl-mayastor upgrade -n openebs --set 'io_engine.coreList={2,3}'
```

and job completed:
```
Application 'upgrade' revision d0a6618f4898 (v2.7.0+0)
  2024-10-28T13:02:47.386848Z  INFO upgrade_job: Validated all inputs
    at k8s/upgrade/src/bin/upgrade-job/main.rs:64

  2024-10-28T13:02:47.725892Z  INFO upgrade_job::helm::upgrade: Verified that openebs helm chart release 'openebs' has version 4.0.1
    at k8s/upgrade/src/bin/upgrade-job/helm/upgrade.rs:359

  2024-10-28T13:02:47.915995Z  INFO upgrade_job::upgrade::data_plane: Skipping data-plane upgrade: All data-plane Pods are already upgraded
    at k8s/upgrade/src/bin/upgrade-job/upgrade/data_plane.rs:43
```

but io_engines are still running on 1,2 cpus as before any changes.

I had also did helm upgrade with:
```
io_engine:
    coreList: [2,3]
```
and also no change.

I did also kill -9 io_engine pid and it was also recreated on 1,2 cpus. 

So not sure if that setting is working at all, maybe should just keep defaults 1,2 and isolate these cpus.

**Comment by tiagolobocastro** [comment]
It is `--set` IIRC but actually that isn't going to work unless is done during the install. For post-install it needs the upgrade changes which @niladrih is working on.
@niladrih can you confirm here?

**Comment by niladrih** [comment]
That's right, I should be able to merge https://github.com/openebs/mayastor-extensions/pull/553 soon.

**Comment by tiagolobocastro** [comment]
@todeb so for now you'd need to use helm upgrade yourself, rather than kubectl-mayastor upgrade
And then you'd have to gracefully delete the io-engine pods one by one

**Comment by Abhinandan-Purkait** [comment]
@niladrih Is this meant to go in 2.8? If so please assign the milestone.

**Comment by niladrih** [comment]
This has been included in v2.7.2 (same-version upgrade/configuration update)
